vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -30,11 +30,52 @@ function DownloadFileWithRetry {
         $retryDelay = 0
     )
     curl.exe -f --retry $retryCount --retry-delay $retryDelay -L $URL -o $Dest
-    if (-not $?) {
+    if ($LASTEXITCODE) {
         throw "Curl exited with '$LASTEXITCODE' while attemping to download '$URL'"
     }
 }
 
+function Retry-Command {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Position=0, Mandatory=$true)]
+        [scriptblock]$ScriptBlock,
+
+        [Parameter(Position=1, Mandatory=$true)]
+        [string]$ErrorMessage,
+
+        [Parameter(Position=2, Mandatory=$false)]
+        [int]$Maximum = 5,
+
+        [Parameter(Position=3, Mandatory=$false)]
+        [int]$Delay = 10
+    )
+
+    Begin {
+        $cnt = 0
+    }
+
+    Process {
+        do {
+            $cnt++
+            try {
+                $ScriptBlock.Invoke()
+                if ($LASTEXITCODE) {
```

Comment:
FYI: @marosset @jsturtevant `docker pull` fails without throwing exception so we need to check $LASTEXITCODE after it.

Reply:
Unfortunately neither `$?` or `$LASTEXITCODE` do not seem to work as desired in all situations.
It depends on the behaviors of the executable unfortunately.

Reply:
Thanks for the info. I tested $LASTEXITCODE and it works if docker pull a non-exit image.